src/main/java/Duke.java

```diff
@@ -1,10 +1,276 @@
+    InvalidDoneException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! Done index is invalid.";
+    }
+}
+
+class Todo extends Task {
+    Todo(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[T]";
+    }
+}
+
+class Event extends Task {
+    Event(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[E]";
+    }
+}
+
+class Deadline extends Task {
+    Deadline(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[D]";
+    }
+}
+
+class Convert {
+    static String at(String s) {
+        String first = s.split("/at")[0];
+        String second = s.split("/at")[1];
+        return first + "(at:" + second + ")";
+    }
+
+    static String by(String s) {
+        String first = s.split("/by")[0];
+        String second = s.split("/by")[1];
+        return first + "(by:" + second + ")";
+    }
+}
+
 public class Duke {
+    // constant SPACE and LINE for format purposes
+    public static String SPACE = "     ";
+    public static String LINE = "____________________________________________________________\n";
+    // add outer frame lines
+    public static String format(String input) {
+        return SPACE + LINE +
+               SPACE + " " + input + "\n" +
+               SPACE + LINE;
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // greeting and exit messages strings
+        // list and mark strings
+        final String messageHello = "Hello! I'm Duke\n" + SPACE + " " + "What can I do for you?";
+        final String messageBye = "Bye. Hope to see you again soon!";
+        final String messageList = "Here are the task(s) in your list:\n";
+        final String messageMarked = "Nice! I've marked this task as done:\n";
+        final String messageAdded = "Got it. I've added this task:\n";
+        final String messageDelete = "Noted. I've removed this task:\n";
+
+        // set up scanner
+        Scanner scanner = new Scanner(System.in);
+
+        // set up empty list of things to do
+        List<Task> list = new ArrayList<>();
+
+        // start to serve
+        System.out.println(format(messageHello));
+
+        // continue if have more commands (that are not "bye")
+        // echo the command or say bye
+        while (scanner.hasNextLine()) {
+            String currentCommand = scanner.nextLine();
+            String priorCommand = currentCommand.split(" ")[0];
+
+            if (currentCommand.equals("bye")) {
```

Comment:
It would be good to make this portion of code more readable! For example, maybe you can consider using switch statements in place of the if-else statements, and consider abstracting the relevant logic into separate methods?